File: javascript/programmers/javascript/level2/카펫.js
const brown = 24;
const yellow = 24;

const solution = (brown, yellow) => {
  const answer = [];
  for (let i = 1; i <= yellow; i++) {
    if (yellow % i !== 0) {
      continue;
    }
    //console.log(i);
    if ((yellow / i) * 2 + i * 2 === brown - 4) {
      answer.push(yellow / i + 2, i + 2);
      break;
    }
  }
  return answer;
};

console.log(solution(brown, yellow));

File: python/programmers/level2/카펫.py
def solution(brown, yellow):
    if yellow == 1:
        return [3, 3]
    for i in range(1, yellow):
        if yellow / i == yellow // i and i * 2 + yellow // i * 2 + 4 == brown:
            return [yellow // i + 2, i + 2]


print(solution(24, 24))
